client/my-sites/plans-features-main/hooks/use-is-custom-domain-allowed-on-free-plan.ts

```diff
@@ -1,22 +1,21 @@
-import { getTld } from 'calypso/lib/domains';
 import { useExperiment } from 'calypso/lib/explat';
 import type { DataResponse } from 'calypso/my-sites/plan-features-2023-grid/types';
 
 const useIsCustomDomainAllowedOnFreePlan = (
 	flowName?: string | null,
-	domainName?: string
+	hasPaidDomainName?: boolean
 ): DataResponse< boolean > => {
-	const [ isLoadingAssignment, experimentAssignment ] = useExperiment(
-		'calypso_onboarding_plans_dotblog_on_free_plan_202307',
-		{
-			isEligible:
-				flowName === 'onboarding' && domainName != null && getTld( domainName ) === 'blog',
-		}
-	);
+	const EXPERIMENT_NAME =
+		flowName === 'onboarding'
+			? 'calypso_onboarding_plans_paid_domain_on_free_plan'
+			: 'calypso_onboardingpm_plans_paid_domain_on_free_plan';
+	const [ isLoading, assignment ] = useExperiment( EXPERIMENT_NAME, {
```

Comment:
When does experiment assignment occur? When they click the free plan button?

Reply:
This hooks run when this hooks runs on the plans page.
https://github.com/Automattic/wp-calypso/blob/98b69c5f971499d4fbaec0f55984c8adcd1d0392/client/my-sites/plans-features-main/index.tsx#L231-L234

In other words the assignment will happen as soon as the plans page loads.

Reply:
Awesome, thanks for the info!. Setting the assignment on plan page load works well IMO!